File: python/person_test.py
import unittest
from person import Person

class PersonTest(unittest.TestCase):
    def setUp(self):
        self.jeff = Person('Jeff')
        self.susy = Person('Susy')

    def test_name(self):
        self.assertEqual(self.jeff.name, 'Jeff')

    def test_greet_hello_jeff(self):
        self.assertEqual(self.jeff.greet('Hello'), "Hi, I'm Jeff")

    def test_greet_hello_susy(self):
        self.assertEqual(self.susy.greet('Hello'), "Hi, I'm Susy")

    def test_greet_goodbye(self):
        self.assertEqual(self.jeff.greet('Goodbye'), 'Bye')

    def test_greet_other(self):
        self.assertEqual(self.jeff.greet('Yo'), 'Yo')

if __name__ == '__main__':
    unittest.main()

File: ruby/person_test.rb
require 'minitest/test'
require 'minitest/autorun'
require File.expand_path('../person', __FILE__)

class PersonTest < MiniTest::Test
  def setup
    @jeff = Person.new('Jeff')
    @susy = Person.new('Susy')
  end

  def test_name
    assert_equal 'Jeff', @jeff.name
  end

  def test_greet_hello_jeff
    assert_equal "Hi, I'm Jeff", @jeff.greet('Hello')
  end

  def test_greet_hello_susy
    assert_equal "Hi, I'm Susy", @susy.greet('Hello')
  end

  def test_greet_goodbye
    assert_equal 'Bye', @jeff.greet('Goodbye')
  end

  def test_greet_other
    assert_equal 'Yo', @jeff.greet('Yo')
  end
end
